docs/source/cpp/getting_started.rst

```diff
@@ -27,6 +27,7 @@ User Guide
    conventions
    build_system
    memory
+   threading
```

Comment:
I'd move this much farther below in the table of contents, since this is more of an advanced topic that only comes up when using certain APIs. What do you think?

Reply:
I moved it down to just past the gdb section.